XenAdmin/Controls/CustomDataGraph/ArchiveMaintainer.cs

```diff
@@ -116,67 +156,65 @@ private void StartUpdateLoop(object _)
             var serverWas = ServerNow;
             InitialLoad(serverWas);
 
-            while (!_cancellationTokenSource.Token.IsCancellationRequested)
+            while (!RequestedCancellation)
             {
-                if (serverWas - LastFiveSecondCollection > TimeSpan.FromSeconds(5))
+                if (ServerNow - LastFiveSecondCollection > TimeSpan.FromSeconds(5))

```

Comment:
Was the replacement of serverWas with ServerNow intentional? Every thime this is called, it calls DateTime.UtcNow, which returns a different time. Don't we need to go through this if blocks with the same time?

Reply:
oops I was playing with it and forgot about it. needs tweaking